### PR TITLE
PMM-5692 - Auto UI tests for Basic and Enhanced Metric Collection for RDS Instances

### PR DESCRIPTION
**Pull request by IvanaWzientkova**
Verify disabling enhanced metrics for rds - Done
Verify adding RDS instances - Done
Verify disabling basic metrics for RDS - Done

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/grafana-dashboards?pullRequest=715) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/grafana-dashboards?pullRequest=715) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/grafana-dashboards?pullRequest=715) it.</sub>

File: pmm-app/tests/verifyAWSRDSMySQLInstance_test.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ Before(async I => {
 });
 
 Scenario(
-  'Verify Discovery and adding AWS RDS MySQL 5.6 instance for monitoring @not-pr-pipeline',
+  'PMM-T138 - Verify disabling enhanced metrics for RDS',
```
Comment (puneet0191) [inline pmm-app/tests/verifyAWSRDSMySQLInstance_test.js:8]: Add @not-pr-pipeline

File: pmm-app/tests/pages/homePage.js [hunk 2]
```diff
@@ -44,4 +44,10 @@ module.exports = {
       'Update operation failed'
     );
   },
+
+    async verifyVisibleService(serviceName) {
+    I.scrollPageToBottom();
+    let serviceExists = "//div[@class='react-grid-item']/descendant::p[contains(text(),'" + serviceName + "')]";
```
Comment (puneet0191) [inline pmm-app/tests/pages/homePage.js:50]: const

File: pmm-app/tests/pages/overviewPage.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const { I, overviewPage } = inject();
```
Comment (puneet0191) [inline pmm-app/tests/pages/overviewPage.js:1]: We use dashboardPage.js it has all the pages as part the module, please try to use the same

File: pmm-app/tests/pages/pmmInventoryPage.js [hunk 4]
```diff
@@ -58,6 +59,32 @@ module.exports = {
     }
   },
 
+  async verifyMetricsFlags(serviceName) {
+    let servicesLink = this.fields.pmmServicesLocator;
```
Comment (puneet0191) [inline pmm-app/tests/pages/pmmInventoryPage.js:63]: use `const` please

File: pmm-app/pr.codecept.js [hunk 5]
```diff
@@ -27,7 +27,8 @@ exports.config = {
     amiInstanceSetupPage: './tests/pages/amiInstanceSetupPage.js',
     pmmSettingsPage: './tests/pages/pmmSettingsPage.js',
     mysqlTableDetailsPage: './tests/pages/mysqlTableDetailsPage.js',
-    dashboardPage: './tests/pages/dashboardPage.js'
+    dashboardPage: './tests/pages/dashboardPage.js',
```
Comment (puneet0191) [inline pmm-app/pr.codecept.js:30]: overviewPage should already be part of dashboardPage, if not please add, just follow how we did for others.

File: pmm-app/tests/pages/dashboardPage.js [hunk 6]
```diff
@@ -406,4 +412,11 @@ module.exports = {
   waitForDashboardOpened() {
     I.waitForElement(this.fields.metricTitle, 30);
   },
+
+  async verifyExisitngServiceName(serviceName)
+  {
+      I.click(this.mySQLInstanceOverview.serviceName);
+      let existingFilter = "//span[@class='variable-option-icon']/following-sibling::span[contains(text(), '" + serviceName + "')]"
```
Comment (atymchuk) [inline pmm-app/tests/pages/dashboardPage.js:419]: `existingFilter` should be a const.

File: pmm-app/tests/pages/pmmInventoryPage.js [hunk 7]
```diff
@@ -58,6 +59,32 @@ module.exports = {
     }
   },
 
+  async verifyMetricsFlags(serviceName) {
+    const servicesLink = this.fields.pmmServicesLocator; 
+    const agentLinkLocator = this.fields.agentsLink;
+    I.waitForElement(servicesLink,20);
+    I.click(servicesLink);
+    let nodeId = await this.getNodeId(serviceName);
```
Comment (atymchuk) [inline pmm-app/tests/pages/pmmInventoryPage.js:67]: `nodeId` should be a const

File: pmm-app/tests/pages/pmmInventoryPage.js [hunk 8]
```diff
@@ -58,6 +59,32 @@ module.exports = {
...
+     "//span[contains(text(), '" +
+       nodeId +
+        "')]/following-sibling::span[contains(text(),'basic_metrics_disabled: true')]"
+    I.seeElement(flagExists);
+  },
+
+  async getNodeId(serviceName)
+  {
+    const nodeIdLocator = this.fields.serviceIdLocatorPrefix + serviceName + "')]/following-sibling::td[1]";
+    const nodeId = await I.grabTextFrom(nodeIdLocator);
```
Comment (atymchuk) [inline pmm-app/tests/pages/pmmInventoryPage.js:84]:
```suggestion
    return await I.grabTextFrom(nodeIdLocator);
```

File: pmm-app/tests/pages/pmmInventoryPage.js [hunk 9]
```diff
@@ -15,6 +15,7 @@ module.exports = {
     nodesLinkOld: "//a[contains(text(), 'Nodes')]",
     pmmAgentLocator: "//table//td[contains(text(), 'PMM Agent')]",
     serviceIdLocatorPrefix: "//table//tr/td[3][contains(text(),'",
+    pmmServicesLocator: "//div[@role='tab'][contains(text(),'Services')]",
```
Comment (yurkovychv) [inline pmm-app/tests/pages/pmmInventoryPage.js:18]: please rename it to pmmServicesSelector

**Comment by puneet0191** [comment]
Will test changes on Job tomorrow and merge it after review comments addressed thanks a lot.

File: pmm-app/tests/pages/dashboardPage.js [hunk 10]
```diff
@@ -296,6 +296,13 @@ module.exports = {
       'MySQL Table Definition Cache',
     ],
   },
+
+  mySQLInstanceOverview: {
+    url: 'graph/d/mysql-instance-overview/mysql-instances-overview?orgId=1&from=now-2m&to=now&refresh=1m',
+    serviceName:
+      "//label[contains(text(), 'Service Name')]/following-sibling::value-select-dropdown/descendant::a[@class='variable-value-link']",
```
Comment (atymchuk) [inline pmm-app/tests/pages/dashboardPage.js:303]: This looks a bit too complicated according to the analyzer https://davertmik.github.io/locator

File: pmm-app/tests/pages/qanPage.js [hunk 11]
```diff
@@ -137,13 +137,18 @@ module.exports = {
 
   getFilterLocator(filterValue) {
     const filterLocator =
-     "//section[@class='aside__filter-group']//span[contains(text(), '" +
-     filterValue +
-     "')]/../span[@class='checkbox-container__checkmark']";
+      "//section[@class='aside__filter-group']//span[contains(text(), '" +
```
Comment (atymchuk) [inline pmm-app/tests/pages/qanPage.js:140]: The general impression I get when reading this PR is that our locators are not sustainable and not resistant to change, so any minor change  in html (with no change in the component logic) would require to update the locators. We should just leverage `data-qa` or `data-test-id` properties. I would encourage to upgrade our components to provide a way to universally expose those props to make it possible.

The other best practice is to prefer css-selectors over xml-selectors as being more resilient.

Comment (atymchuk) [reply pmm-app/tests/pages/qanPage.js:140]: Good example: https://github.com/percona/grafana-dashboards/pull/715
No logic changed there, but the tests failed.

**Pull request merged**